packages/react-components/react-infobutton/src/components/InfoButton/useInfoButton.tsx

```diff
@@ -74,5 +82,43 @@ export const useInfoButton_unstable = (props: InfoButtonProps, ref: React.Ref<HT
   state.popover.open = popoverOpen;
   state.popover.onOpenChange = mergeCallbacks(state.popover.onOpenChange, (e, data) => setPopoverOpen(data.open));
 
+  const closeOnTabBack = (ev: KeyboardEvent) => {
+    ev.stopPropagation();
+    if (ev.currentTarget === ev.target && ev.shiftKey && ev.key === 'Tab') {
+      setPopoverOpen(false);
+    }
+  };
+  const closeOnTabForward = (ev: KeyboardEvent) => {
```

Comment:
I think you'd have an easier time using the `focusout` event listener with a callback ref. That way you'd just need to check that as soon as focus leaves the popover, then it should close:

```ts
const focusOutRef = React.useCallback((el) => {
  if (!el) {
    return;
  }

  el.addEventListener('focusout', (e) => {
    const lostFocus = e.target
    const nextFocused = e.relatedTarget

    if (!elementContains(lostFocus, nextFocused)) {
      /* only run when focus is lost from the popover */
    } 
  })
}, []);
```